### PR TITLE
#162948872 password reset

### PR DESCRIPTION
**Pull request by jamesbeamie**
#### What does this PR do?

- Allows a user to request for a password reset
- Provides the user with a password reset link

#### Description of Task to be completed?

* User should enter their emails to be sent a password reset link 
* User should be directed to a page to enter and confirm their new password
#### How should this be manually tested?
* Click forgot password link on the login modal
image

* Enter your registered email
image

#### Reset password
* Go to your email and click the link sent for resetting the password.
* Enter the new password and confirm it
image

#### What are the relevant pivotal tracker stories?
[#162948872 ](https://www.pivotaltracker.com/story/show/162948872)

### CODE REVIEW

**Review by danuluma — APPROVED** [review]

File: src/actions/ForgotPassword.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    handleChange = (e) => {
+      this.setState({
+        email: e.target.value
+      });
+    }
+
+    handleSubmit = (e) => {
+      e.preventDefault();
+      const email_data = this.state.email;
```
[the remark this answers is not in the document]
Comment (Betty-Kebenei) [reply src/actions/ForgotPassword.js]: fix this kindly

[the remark this answers is not in the document]
Comment (Betty-Kebenei) [reply src/actions/ForgotPassword.js]: For js underscores are not recommended

File: src/actions/ResetPassword.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { Form, Message } from "semantic-ui-react";
+import React, { Component } from "react";
+import axiosConfig from "./axiosConfig";
+
+class ResetPassword extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: null, confirm_password: null, success: null };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (Betty-Kebenei) [inline src/actions/ResetPassword.js]: you don't have to bind functions if you have used arrow functions

**Review by Deekerubo — COMMENTED** [review]
Add _snapshots_ to .gitignore

File: source.env [hunk 3]
```diff
@@ -0,0 +1 @@
+BASE_URL="https://ah-alpha.herokuapp.com/"
```
Comment (Betty-Kebenei) [inline source.env]: Delete this file in git and locally

File: src/actions/ForgotPassword.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { Form, Message } from "semantic-ui-react";
```
Comment (Betty-Kebenei) [inline src/actions/ForgotPassword.js]: This is a component, it should be in the components' folder. And create actions for it.

File: src/actions/ResetPassword.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import { Form, Message } from "semantic-ui-react";
```
Comment (Betty-Kebenei) [inline src/actions/ResetPassword.js]: This too is a component

**Review by Deekerubo — APPROVED** [review]

File: src/assets/css/index.scss [hunk 6]
```diff
@@ -12,4 +12,3 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
-
```
Comment (Betty-Kebenei) [inline src/assets/css/index.scss:15]: Add this line back

File: src/assets/js/main.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Betty-Kebenei) [inline src/assets/js/main.js]: Delete this file in git and locally

File: src/components/Home.js [hunk 8]
```diff
@@ -1,8 +1,11 @@
 import React from "react";
```
Comment (Betty-Kebenei) [inline src/components/Home.js]: Wachira had fixed this

**Review by Ruiru11 — APPROVED** [review]
looks Good

**Pull request merged**